internal/services/providers/eks/client/aws.go

```diff
@@ -253,3 +250,44 @@ func (c *client) GetInstancesByInstanceIDs(ctx context.Context, instanceIDs []st
 
 	return instances, nil
 }
+
+func (c *client) getMetadataWithContext(ctx context.Context) (string, error) {
+	ctx, cancel := c.requestContext(ctx)
+	defer cancel()
+
+	c.log.Debug("fetching EC2 instance metadata")
+	return c.metaClient.GetMetadataWithContext(ctx, "instance-id")
+}
+
+func (c *client) describeInstancesWithContext(ctx context.Context, req *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	ctxWithTimeout, cancel := c.requestContext(ctx)
+	defer cancel()
+
+	c.log.Debug("fetching EC2 instance information")
+	return c.ec2Client.DescribeInstancesWithContext(ctxWithTimeout, req)
+}
+
+func (c *client) requestContext(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
I wonder if this is handled correctly. Timeout should be set in API HTTP client, otherwise the context cancel will be ignored by http request in progress. Definitely add an actual test where API is simulated by http server with a delay bigger than a configured timeout.